maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -816,6 +894,81 @@ private void assembleInheritance( List<ModelData> lineage, ModelBuildingRequest
             inheritanceAssembler.assembleModelInheritance( child, parent, request, problems );
         }
     }
+    
+    private InputStream transformData( final Path pomFile )
+                    throws IOException, TransformerException, SAXException, ParserConfigurationException
+    {
+        final TransformerFactory transformerFactory = Factories.newTransformerFactory() ;
+        
+        final PipedOutputStream pipedOutputStream  = new PipedOutputStream();
+        final PipedInputStream pipedInputStream  = new PipedInputStream( pipedOutputStream );
+
+        // Should always be FileSource for reactor poms
+        // System.out.println( "transforming " + pomFile );
+        
+        final SAXSource transformSource =
+            new SAXSource( buildPomXMLFilterFactory.get().get( pomFile ),
+                           new org.xml.sax.InputSource( new FileInputStream( pomFile.toFile() ) ) );
+
+        OutputStream out;
+        if ( xmlFilterListener != null )
+        {
+            out = new FilterOutputStream( pipedOutputStream )
+            {
+                @Override
+                public void write( byte[] b, int off, int len )
+                    throws IOException
+                {
+                    super.write( b, off, len );
+                    xmlFilterListener.write( pomFile, b, off, len );
+                }  
+            };
+        }
+        else
+        {
+            out = pipedOutputStream;
+        }
+
+        final StreamResult result = new StreamResult( out );
+        
+////        final Callable<Void> callable = () ->
```

Comment:
Se should remove commented lines

Reply:
Now with the improved caching in place, I should be able to make use of it.
thanks for the reminder of these comments, I need to improve the exception handling here too.